src/components/common/form/InputWithAdornment.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+import React, { useCallback, useEffect } from 'react'
+import { isMobile, isMobileSafari } from 'mobile-device-detect'
+import { StyleSheet, TextInput, TouchableOpacity, View } from 'react-native'
+import normalize from '../../../lib/utils/normalizeText'
+import { useCurriedSetters } from '../../../lib/undux/SimpleStore'
+import { withStyles } from '../../../lib/styles'
+import Icon from '../view/Icon'
+import Config from '../../../config/config'
+import ErrorText from './ErrorText'
+
+const InputText = ({
+  showAdornment,
+  adornment,
+  adornmentAction,
+  adornmentSize = 16,
+  adornmentStyle,
+  error,
+  styles,
+  theme,
+  style,
+  getRef,
+  onBlur,
+  ...props
+}) => {
+  const [setMobileSafariKeyboardShown, setMobileKeyboardShown] = useCurriedSetters([
+    'isMobileSafariKeyboardShown',
+    'isMobileKeyboardShown',
+  ])
+  const shouldChangeSizeOnKeyboardShown = isMobileSafari && Config.safariMobileKeyboardGuidedSize
```

Comment:
this is props & state independent, could be moved out of the component

Reply:
done